test/bitset.jl

```diff
@@ -351,3 +351,12 @@ end
     # union! with an empty range doesn't modify the BitSet
     @test union!(x, b:a) == y
 end
+
+@testset "union!(::BitSet, ::AbstractUnitRange) when two ranges do not overlap" begin
```

Comment:
These tests seem to pass on e.g. 1.8-rc1 so it seems they do not cover the fixed behavior?

Reply:
@KristofferC
I find that the failure of original code can only be triggered occasionally. (Maybe `Base._growbeg!/_growend!` allocates uninitialized memory that can be zero by accident)
I will change the code so that it can trigger failure consistently.

Reply:
I test the new code in 1.8-rc1 (Win and Linux), and it triggers failure immediately.